jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -38,22 +38,32 @@
 import org.apache.http.impl.client.DefaultHttpClient;
 
 /**
- * Thread-safe HTTP client with the automatic insecure connection failover feature. Intended to be
- * created once and shared to be called at multiple places. Callers should close the returned {@link
- * Response}.
+ * Thread-safe HTTP client that can automatically failover from secure HTTPS to insecure HTTPS or
+ * HTTP. Intended to be created once and shared to be called at multiple places. Callers should
+ * close the returned {@link Response}.
  *
  * <p>The failover (if enabled) in the following way:
  *
  * <ul>
- *   <li>When port was given:
+ *   <li>When a port is provided (for example {@code my-registry:5000/my-repo}):
  *       <ol>
  *         <li>Attempts secure HTTPS.
- *         <li>If it fails due to {@link SSLException}, attempts insecure HTTPS.
- *         <li>If it fails again due to {@link SSLException}, attempts plain-HTTP.
+ *         <li>If (1) fails due to {@link SSLException}, attempts insecure HTTPS (disables
+ *             certificate validation).
+ *         <li>If (2) fails again due to {@link SSLException}, attempts plain-HTTP.
  *       </ol>
- *   <li>When port was not given: follows same execution path above, but additionally, if the very
- *       first secure HTTPS (at port 443) fails due to non-timeout {@link ConnectException},
- *       attempts plain-HTTP at port 80.
+ *   <li>When a port is not provided (for example {@code my-registry/my-repo}):
+ *       <ol>
+ *         <li>Attempts secure HTTPS.
+ *         <li>If (1) fails due to {@link SSLException}, attempts insecure HTTPS (disables
+ *             certificate validation).
+ *         <li>If (2) fails again due to {@link SSLException}, attempts plain-HTTP.
+ *         <li>Or, if (1) fails due to non-timeout {@link ConnectException}, attempts plain-HTTP at
+ *             port 80.
+ *       </ol>
+ *       In sum, this follows the same execution path as when a port is provided, but additionally,
+ *       it can fall back to HTTP at port 80 when the very first secure HTTPS (at port 443) fails
+ *       when it cannot connect at all.
```

Comment:
I found this text (and the previous text) left me wondering if we attempt plain HTTP on port 443.  I think being explicit about the ports in the actual steps is better, and then we can drop this text.

Reply:
Ah, yeah, you're right. That's the difference. We don't attempt plain HTTP on port 443 in this case. I've dropped the summary text.